File: lab 1 scala/src/zadanie2.scala


class zadanie2 {
  
  
  def twoLast[A](xss:List[A]):(A,A)=
    		if (xss==Nil) throw new Exception("pusta")
		else if (xss.tail == Nil) throw new Exception("tylko jeden")
		else if ( xss.tail.tail == Nil) (xss.head , xss.tail.head )
		else twoLast(xss.tail)
	
twoLast(List(1,2,3,4,5,6,7))
  
  
  
  
  
  
  
  
  
}

File: lab 1 ocaml/zadanie2.ml
(*zadanie 2*)
let rec twoLast xss=
		if xss=[] then failwith "pusta"
		else if List.tl xss = [] then failwith "tylko jeden"
		else if  List.tl(List.tl xss) = [] then (List.hd xss , List.hd(List.tl xss) )
		else twoLast (List.tl xss);;
	
twoLast[1;3;4;5;7]